packages/origin-ui-core/src/components/devices/ProducingDeviceTable.tsx

```diff
@@ -57,20 +57,19 @@ export function ProducingDeviceTable(props: IOwnProps) {
     const user = useSelector(getUserOffchain);
     const producingDevices = useSelector(getProducingDevices);
     const baseURL = useSelector(getBaseURL);
-    const organizations = useSelector(getOrganizations);
     const environment = useSelector(getEnvironment);
+    const offChainDataSource = useSelector(getOffChainDataSource);
 
     const dispatch = useDispatch();
 
-    function enrichProducingDeviceData(): IEnrichedProducingDeviceData[] {
+    async function enrichProducingDeviceData(): Promise<IEnrichedProducingDeviceData[]> {
+        const deviceClient = offChainDataSource.deviceClient;
         const enriched: IEnrichedProducingDeviceData[] = [];
-
         for (const device of producingDevices) {
-            const organization = organizations.find((o) => o.id === device.organization);
-
+            const deviceWithRelations = await deviceClient.getById(device.id, false);
             enriched.push({
                 device,
-                organizationName: organization?.name,
+                organizationName: (deviceWithRelations as any)?.organization.name,
```

Comment:
This should already be contained in the `device` variable. If it's not, add `loadRelationIds` to `deviceClient.getAll()`. Re-fetching every device with a separate API call is a performance downgrade.